src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -23,16 +24,30 @@ class PropertyAccessor implements PropertyAccessorInterface
 {
     const VALUE = 0;
     const IS_REF = 1;
+    const ACCESS_TYPE = 0;
+    const ACCESS_NAME = 1;
+    const ACCESS_REF = 2;
+    const ACCESS_ADDER = 3;
+    const ACCESS_REMOVER = 4;
+    const ACCESS_TYPE_METHOD = 0;
+    const ACCESS_TYPE_PROPERTY = 1;
+    const ACCESS_TYPE_ADDER_AND_REMOVER = 2;
+    const ACCESS_TYPE_NOT_FOUND = 3;
 
     private $magicCall;
+    private $cache;
+    private $propertyPathCache = array();
+    private $readPropertyCache = array();
+    private $writePropertyCache = array();
 
     /**
      * Should not be used by application code. Use
      * {@link PropertyAccess::createPropertyAccessor()} instead.
      */
-    public function __construct($magicCall = false)
+    public function __construct($magicCall = false, Cache $cache = null)
```

Comment:
This is a BC break, because we already have a second argument in newer branches, and it is not a Cache instance